builder: upgrade to autouseradd 1.2.0

benesch opened the pull request:
The new version of autouseradd has a fix for the TMPDIR propagation
issue. (It is also a bit more careful to propagate other variables that
are normally dropped in "secure-execution mode", on the off chance that
a later version of the builder wants to propagate those.)

@tbg up to you whether you want this. You'll have to rebuild the builder since I no longer have Docker Hub permissions (and also don't feel like sitting through the 1hr long build process :D). But should be quick if you've got a warm Docker cache already.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/51076)
<!-- Reviewable:end -->

tbg commented:
Thanks! I'll take it but will hold off until I have my Linux desktop computer around.

benesch committed:
builder: upgrade to autouseradd 1.2.0

The new version of autouseradd has a fix for the TMPDIR propagation
issue. (It is also a bit more careful to propagate other variables that
are normally dropped in "secure-execution mode", on the off chance that
a later version of the builder wants to propagate those.)

Release note: None

tbg commented:
Rebased this PR - `builder.sh init` barfed on the first run, but I suspect it's due to a download failure, currently running the second iteration.

tbg commented:
Yeah not sure. This isn't exactly helpful:

```
[INFO ]  Performing some trivial sanity checks
[INFO ]  Build started 20200814.144039
[INFO ]  Building environment variables
[EXTRA]  Preparing working directories
[EXTRA]  Installing user-supplied crosstool-NG configuration
[EXTRA]  =================================================================
[EXTRA]  Dumping internal crosstool-NG configuration
[EXTRA]    Building a toolchain for:
[EXTRA]      build  = x86_64-pc-linux-gnu
[EXTRA]      host   = x86_64-pc-linux-gnu
[EXTRA]      target = x86_64-w64-mingw32
[EXTRA]  Dumping internal crosstool-NG configuration: done in 0.03s (at 00:01)
[INFO ]  =================================================================
[INFO ]  Retrieving needed toolchain components' tarballs
[EXTRA]    Retrieving 'mingw-w64-v5.0.1'
[ERROR]   
[ERROR]  >>
[ERROR]  >>  Build failed in step 'Retrieving needed toolchain components' tarballs'
[ERROR]  >>        called in step '(top-level)'
[ERROR]  >>
[ERROR]  >>  Error happened in: do_libc_get[scripts/build/libc/mingw.sh@869]
[ERROR]  >>        called from: main[scripts/crosstool-NG.sh@607]
[ERROR]  >>
[ERROR]  >>  For more info on this error, look at the file: 'build.log'
[ERROR]  >>  There is a list of known issues, some with workarounds, in:
[ERROR]  >>      '/usr/local/ct-ng/share/doc/crosstool-ng/crosstool-ng-1.23.0/B - Known issues.txt'
[ERROR]  >>
[ERROR]  >>  If you feel this is a bug in crosstool-NG, report it at:
[ERROR]  >>      https://github.com/crosstool-ng/crosstool-ng/issues/
[ERROR]  >>
[ERROR]  >>  Make sure your report includes all the information pertinent to this issue.
[ERROR]  >>  Read the bug reporting guidelines here:
[ERROR]  >>      http://crosstool-ng.github.io/support/
[ERROR]   
[ERROR]  (elapsed: 1:42.95)
[01:43] / /usr/local/ct-ng/bin/ct-ng:147: recipe for target 'build' failed

```

tbg commented:
The image was pushed, so hopefully this is going to turn green now.

tbg commented:
bors r+
